From 47c5147e57318333ea446f2cb568a992b23f082f Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sun, 11 Jan 2015 19:24:43 -0800
Subject: [PATCH 4/9] libunwind: Only include <Availability.h> on Leopard and
 newer OS X versions

http://www.llvm.org/bugs/show_bug.cgi?id=22203

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 15cc2c11afe656d450cfd6f80266491aaf096960)
---
 include/libunwind.h | 22 ++++++++++++++++++----
 src/config.h        |  9 ++++++++-
 2 files changed, 26 insertions(+), 5 deletions(-)

diff --git a/include/libunwind.h b/include/libunwind.h
index 4a75a7b..315a802 100644
--- a/include/libunwind.h
+++ b/include/libunwind.h
@@ -20,12 +20,26 @@
 #include <stddef.h>
 
 #ifdef __APPLE__
-  #include <Availability.h>
-    #ifdef __arm__
-       #define LIBUNWIND_AVAIL __attribute__((unavailable))
+  #if __clang__
+    #if __has_include(<Availability.h>)
+      #include <Availability.h>
+    #endif
+  #elif __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
+    #include <Availability.h>
+  #endif
+
+  #if __arm__
+     #define LIBUNWIND_AVAIL __attribute__((unavailable))
+  #elif defined(__OSX_AVAILABLE_STARTING)
+    #define LIBUNWIND_AVAIL __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_5_0)
+  #else
+    #include <AvailabilityMacros.h>
+    #ifdef AVAILABLE_MAC_OS_X_VERSION_10_6_AND_LATER
+      #define LIBUNWIND_AVAIL AVAILABLE_MAC_OS_X_VERSION_10_6_AND_LATER
     #else
-      #define LIBUNWIND_AVAIL __OSX_AVAILABLE_STARTING(__MAC_10_6, __IPHONE_5_0)
+      #define LIBUNWIND_AVAIL __attribute__((unavailable))
     #endif
+  #endif
 #else
   #define LIBUNWIND_AVAIL
 #endif
diff --git a/src/config.h b/src/config.h
index 123eef1..1cbf057 100644
--- a/src/config.h
+++ b/src/config.h
@@ -29,7 +29,14 @@
 
 // Platform specific configuration defines.
 #ifdef __APPLE__
-  #include <Availability.h>
+  #if __clang__
+    #if __has_include(<Availability.h>)
+      #include <Availability.h>
+    #endif
+  #elif __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
+    #include <Availability.h>
+  #endif
+
   #ifdef __cplusplus
     extern "C" {
   #endif
-- 
2.6.2

